### ISSUE TITLE
Add Lazy Load / Deferred Load Option to Hierarchy

### ISSUE THREAD

**Issue by Enverex** [Frooxius/NeosPublic#468]
Currently (as far as I'm aware) when you load into a world, everything in the hierarchy, and everything everything in the hierarchy refers to, is downloaded.

What I propose is some way of marking objects on the hierarchy in some way as "only load when required" which would defer their loading until they're actually enabled. This means objects with this option enabled aren't actually downloaded until they're enabled in the hierarchy.

Some example use-cases would be:
A world with multiple versions of things (say sets of trees for seasons or something, or variations of buildings, etc). You could simply set the unused ones to disabled and lazy, that way when users join the world, they wouldn't actually download them at all. Allowing you to easily switch between them when updating the map later.

Could also be used to swap entire areas of maps as a transition / pseudo-chunking mechanism.

Basically, a "do not load this until it's enabled in the heiarchy" option.

**Comment by Frooxius** [comment]
This is planned, we have an issue for this here already: https://github.com/Frooxius/NeosPublic/issues/183 Closing this one as a duplicate. 

It's good suggestion though of course!
